bluesky_queueserver/manager/start_manager.py

```diff
@@ -34,103 +36,71 @@ def _create_conn_pipes(self):
-        self._thread_conn.start()
+    # ======================================================================
+    #             Handlers for messages from RE Manager
 
-    def _receive_packet_thread(self):
-        while True:
-            if self._watchdog_to_manager_conn.poll(0.1):
-                try:
-                    msg = self._watchdog_to_manager_conn.recv()
-                    # Messages should be handled in the event loop
-                    self._conn_received(msg)
-                except Exception as ex:
-                    logger.exception("Exception occurred while waiting for packet: %s" % str(ex))
-                    break
-
-    def _conn_received(self, msg):
-        type, value = msg["type"], msg["value"]
-
-        if type == "notification":
-            if value == "alive":
-                self._register_heartbeat()
-
-        if type == "command":
-            logger.debug("Command received from RE Manager: %s" % pprint.pformat(msg))
-            if value == "start_re_worker":
-                self._start_re_worker()
-                msg_out = {"type": "report", "value": {"msg": msg, "success": True}}
-                self._watchdog_to_manager_conn.send(msg_out)
-
-            if value == "join_re_worker":
-                success = self._join_re_worker()
-                msg_out = {"type": "report", "value": {"msg": msg, "success": success}}
-                self._watchdog_to_manager_conn.send(msg_out)
-
-            if value == "kill_re_worker":
-                self._kill_re_worker()
-                msg_out = {"type": "report", "value": {"msg": msg, "success": True}}
-                self._watchdog_to_manager_conn.send(msg_out)
-
-            if value == "manager_stopping":
-                # Manager informed that it is stopping and should not be restarted
-                self._manager_is_stopping = True
-
-        if type == "request":
-            if value == "is_worker_alive":
-                result = self._is_worker_alive()
-                msg_out = {"type": "result", "value": {"request": value, "result": result}}
-                self._watchdog_to_manager_conn.send(msg_out)
-
-    def _start_re_worker(self):
+    def _start_re_worker_handler(self):
         """
         Creates worker process. This is a quick operation, because it starts RE Worker
         process without waiting for initialization.
         """
         logger.info("Starting RE Worker ...")
-        self._re_worker = RunEngineWorker(conn=self._manager_conn, name="RE Worker Process")
-        self._re_worker.start()
-
-    def _join_re_worker(self):
+        try:
+            self._re_worker = RunEngineWorker(conn=self._manager_conn, name="RE Worker Process")
+            self._re_worker.start()
+            success, err_msg = True, ""
+        except Exception as ex:
+            success, err_msg = False, str(ex)
+        return {"success": success, "err_msg": err_msg}
+
+    def _join_re_worker_handler(self, *, timeout=0.5):
```

Comment:
Just a question for my own benefit, rather than a comment:

This semantic to force keyword-only arguments is used in many places. What is the motivation for restricting to keyword invocations only? Is it just to make sure the calls will still work, even if the function footprint is changed?

Reply:
JSON RPC allows for passing arguments as lists, but I think it is safer to pass them as dictionaries. This way the order of the arguments does not matter. `json-rpc` will not accept requests with arguments as lists if handler has keyword-only arguments. But in general the code is more readable if parameter names are specified in function calls, especially in case when libraries are under development. It is easier to relax the restrictions later.

Reply:
It protects you against adding / removing arguments and it it (arguably) makes the calls clearer (as the keyword name is in line).  I think this is as strength for internal code where it is mostly read and only rarely typed.